src/main/java/org/apache/maven/plugins/dependency/tree/XMLDependencyNodeVisitor.java

```diff
@@ -0,0 +1,186 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.shared.dependency.graph.DependencyNode;
+import org.apache.maven.shared.dependency.graph.traversal.DependencyNodeVisitor;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import java.io.Writer;
+import java.util.List;
+
+/**
+ * A dependency node visitor that serializes visited nodes to
+ * <a href="https://en.wikipedia.org/wiki/XML">XML format</a>
+ *
+ * @author <a href="mailto:sikora.bogdan@webscope.io">Bogdan Sikora</a>
+ * @since 3.1.2
+ */
+public class XMLDependencyNodeVisitor
+    extends AbstractSerializingVisitor
+    implements DependencyNodeVisitor
+{
+    /**
+     * Constructor.
+     *
+     * @param writer the writer to write to.
+     */
+    public XMLDependencyNodeVisitor( Writer writer )
+    {
+        super( writer );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean visit( DependencyNode node )
+    {
+        try
+        {
+            if ( node.getParent() == null || node.getParent() == node )
+            {
+                DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+                Document doc = dBuilder.newDocument();
+                Element rootElement = getNode( doc, node, true );
+                doc.appendChild( rootElement );
+
+                List<DependencyNode> children = node.getChildren();
+                for ( DependencyNode child : children )
+                {
+                    handleChild( doc, child, rootElement );
+                }
+
+                TransformerFactory transformerFactory = TransformerFactory.newInstance();
+                Transformer transformer = transformerFactory.newTransformer();
+                transformer.setOutputProperty( OutputKeys.INDENT, "yes" );
+                transformer.setOutputProperty( "{http://xml.apache.org/xslt}indent-amount", "2" );
+                DOMSource source = new DOMSource( doc );
+
+                StreamResult console = new StreamResult( writer );
+
+                transformer.transform( source, console );
+            }
+        }
+        catch ( ParserConfigurationException | TransformerException e )
+        {
+            e.printStackTrace();
```

Comment:
Think of a better solution: either log with Logger or rethrow.

Reply:
fixed